src/Simulators/MultipleTimedRenders.js

```diff
@@ -0,0 +1,62 @@
+import { createElement, Component } from 'react';
+import { arrayOf, oneOfType, shape, func, number, any } from 'prop-types';
+import SimulatorErrorBoundary from './SimulatorErrorBoundary';
+import scheduleCallbackArgs from './schedule-callback-args';
+
+export default class MultipleTimedRenders extends Component {
+    static propTypes = {
+        initialArgs: oneOfType([arrayOf(any), func]),
+        scheduledArgs: arrayOf(
+            shape({
+                elapsed: number.isRequired,
+                args: oneOfType([arrayOf(any), func]).isRequired
+            }).isRequired
+        ).isRequired,
+        /* (error: Error) => any */
+        onError: func,
+        /* (prop: any) => React.Element<any> */
+        children: func.isRequired
+    };
+
+    constructor(props) {
```

Comment:
No need for a constructor here - class properties will work. They're initialized after the implicit `super` is called in the empty constructor, so you have `this.props` access.

```js
state = {
   args: typeof this.props.initialArgs === 'function' ? initialArgs() : initialArgs
};
```